### DISCUSSION TITLE
Feature Request: Optional Arguments

### DISCUSSION THREAD

**Discussion by m4x74** [chhoumann/quickadd#587] [Ideas]
I would like to use {{VDATE}} as an optional argument, to create tasks with or without a due date. With {{VALUE}} it is possible to skip by just not entering anything. For {{VDATE}} this results in the string `Invalid date`. This feature was basically already described in this [comment on the Obsidian Forum](https://forum.obsidian.md/t/quickadd-plugin/20032/114). 

A more general feature, which would be even better: QuickAdd asking for arguments based on previous arguments (see example below).

`{{VALUE: no due date, with due date}} {{IF: with due date: {{VDATE:Due day?, YYYY-MM-DD}}}}`

**Comment by FeralFlora** [comment]
Hoping this gets fixed soon!

**Comment by Iconca** [comment]
Hi, I think I have a similar request. From what I managed to read it is impossible, but maybe I'm missing something. I wonder if there is a possibility to skip any of my pre-programmed steps. I try to use Quick Add for adding a few specific dataview properties. But I don't always need all of them, thus I usually end up with a bunch of empty fields. They are supposed to be a remainder. A choice. So I would appreciate a function like a "skip this one step" button.  
It looks like this.

```text
- [r] (bookmark_type:: {{VALUE:TO READ:,TO SEE:, TO GO:}}), (item:: {{VALUE:movie,book, event}}),  [title:: {{VALUE:<title>}}], [author:: {{VALUE:<author>}}], [category:: {{VALUE:<category>}}], [publish_date:: {{VALUE:<publish_date>}}],➕ {{date}}
```

So if I know only the title, I get
- [r] (bookmark_type:: TO READ:), (item:: book),  [title:: The Likeness], [author:: ], [category:: ], [publish_date:: ], ➕ 2024-07-06

And I would prefer just this
- [r] (bookmark_type:: TO READ:), (item:: book), [title:: The Likeness],➕ 2024-07-06

I managed only to think of this workaround. 
{{VALUE:<[author:: ]coma>}}
It gives me a prompt which I have to manually rewrite or live empty and click OK, but it defeats a bit the premise of automatization.
I realize It may be easier said than done but I would really appreciate such a function :)

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that.

The concrete request shipped: the `|optional` flag landed in QuickAdd 2.13.0, and it works on `{{VDATE}}`, `{{VALUE}}`, and option lists alike:

```
{{VDATE:Due day?,YYYY-MM-DD|optional}}
```

Submit the prompt empty (or hit Skip) and the token resolves to nothing instead of `Invalid date`. Tip: literal decoration inside the date format disappears with it, e.g. `{{VDATE:due,[📅 ]YYYY-MM-DD|optional}}` drops the emoji too when skipped.

@Iconca: same flag for your bookmark line - `[author:: {{VALUE:author|optional}}]` lets you skip the prompt and leave the value empty. One caveat: literal text *around* a `{{VALUE}}` token (the `[author:: ],` part) stays in the output, so a skipped field renders as `[author:: ]` rather than vanishing entirely. If you want whole segments to disappear conditionally, that's user-script territory in a Macro, where you have full logic available.

Which is also the answer to the generalized `{{IF:...}}` idea: conditional prompting inside the format syntax was deliberately not built - once you need branching, a Macro user script is the better tool and already handles it.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#optional-fields

Closing as resolved.

**Reply by Iconca** [reply]
Hi, @chhoumann 
No worries about late replay. I'll try to use your advice. Thanks for taking interest.
